src/platform_impl/linux/x11/util/geometry.rs

```diff
@@ -183,41 +183,71 @@ impl XConnection {
         window: ffi::Window,
         root: ffi::Window,
     ) -> Result<TranslatedCoords, XError> {
-        let mut translated_coords: TranslatedCoords = unsafe { mem::uninitialized() };
+        let mut x_rel_root = MaybeUninit::uninit();
```

Comment:
I think it should be safe to zero-initialize the whole struct, then pass pointers to the individual members in it.